pypose/module/pf.py

```diff
@@ -187,10 +189,10 @@ def resample_particles(self, q, x):
         r'''
         Resample the set of a posteriori particles
         '''
-        r = torch.rand(self.particles, dtype=x.dtype, device=x.device)
-        cumsumq = torch.cumsum(q, dim=-1)
-        return x[torch.searchsorted(cumsumq, r)]
+        r = torch.rand(x.shape[:-1], dtype=x.dtype, device=x.device)
+        index = torch.searchsorted(torch.cumsum(q, dim=0), r)
+        return x[index, torch.arange(x.shape[1]).unsqueeze(0)] if x.dim()==3 else x[index]
```

Comment:
can we implement this line using `torch.index_select` or `torch.gather` without using `if else`?